### PR TITLE
chore: add sql filter to fetching audit logs

### PR DESCRIPTION
**Pull request by Emyrk**
# What this does

This adds a SQL filter to fetching audit logs. Prior to this change, all audit logs shared the same authorization call, so only 1 was done up front. Now that we have organizational scoped audit logs, listing all audit logs a user can view will operate like workspaces.

The SQL filter is required for pagination, as the filtering has to occur in the database.

### CODE REVIEW

File: coderd/rbac/regosql/configs.go [hunk 1]
```diff
@@ -36,6 +36,21 @@ func TemplateConverter() *sqltypes.VariableConverter {
 	return matcher
 }
 
+func AuditLogConverter() *sqltypes.VariableConverter {
+	matcher := sqltypes.NewVariableConverter().RegisterMatcher(
+		resourceIDMatcher(),
+		sqltypes.StringVarMatcher("COALESCE(audit_logs.organization_id :: text, '')", []string{"input", "object", "org_owner"}),
+		// Templates have no user owner, only owner by an organization.
```
Comment (johnstcn) [inline coderd/rbac/regosql/configs.go:43]: Copy-pasta? This is talking about audit logs, not templates.

File: coderd/rbac/regosql/configs.go [hunk 2]
```diff
@@ -36,6 +36,21 @@ func TemplateConverter() *sqltypes.VariableConverter {
...
+		sqltypes.AlwaysFalse(userOwnerMatcher()),
+	)
+	matcher.RegisterMatcher(
+		// No ACLs on the user type
```
Comment (johnstcn) [inline coderd/rbac/regosql/configs.go:47]: More copy-pasta? What does 'user' mean in this context?

File: coderd/database/querier_test.go [hunk 3]
```diff
@@ -767,6 +773,152 @@ func TestReadCustomRoles(t *testing.T) {
...
+		allLogs = append(allLogs, dbgen.AuditLog(t, db, database.AuditLog{
+			ID:             id,
+			OrganizationID: uuid.Nil,
+		}))
+	}
+
+	orgAuditLogs := map[uuid.UUID][]uuid.UUID{
+		uuid.New(): {uuid.New(), uuid.New()},
+		uuid.New(): {uuid.New(), uuid.New()},
+	}
```
Comment (johnstcn) [inline coderd/database/querier_test.go:795]: This is hard to read, specify what the UUIDs signify in a comment. My intuition is that it's a `map[OrgID]AuditLogID`?

File: coderd/database/querier_test.go [hunk 4]
```diff
@@ -767,6 +773,152 @@ func TestReadCustomRoles(t *testing.T) {
...
+			FriendlyName: "member",
+			ID:           uuid.NewString(),
+			Roles:        rbac.Roles{memberRole},
+			Scope:        rbac.ScopeAll,
+		})
+
+		logs, err := db.GetAuditLogsOffset(siteAuditorCtx, database.GetAuditLogsOffsetParams{})
+		require.NoError(t, err)
+		require.Len(t, logs, 0, "no logs should be returned")
+	})
```
Comment (johnstcn) [inline coderd/database/querier_test.go:841]: Add Given-When-Then comments to clarify the desired behaviour

e.g.

```
// Given: a user who is a member of a single org but has no other roles
// When: the user queries for audit logs
// Then: no results are returned
```

File: coderd/audit_test.go [hunk 5]
```diff
@@ -163,19 +162,18 @@ func TestAuditLogs(t *testing.T) {
 		})
 		require.NoError(t, err)
 
-		// Fetching audit logs without an organization selector should fail
-		_, err = orgAdmin.AuditLogs(ctx, codersdk.AuditLogsRequest{
+		// Fetching audit logs without an organization selector should only
+		// return organization audit logs.
```
Comment (johnstcn) [inline coderd/audit_test.go:166]: for which org?

File: coderd/database/querier_test.go (hunk 3 above)
Comment (johnstcn) [inline coderd/database/querier_test.go:795]: Should we also add some audit logs that are not scoped to any particular org?

File: coderd/rbac/regosql/configs.go (hunk 1 above)
Comment (Emyrk) [reply coderd/rbac/regosql/configs.go:43]: This should be audit logs :+1:

File: coderd/rbac/regosql/configs.go (hunk 2 above)
Comment (Emyrk) [reply coderd/rbac/regosql/configs.go:47]: Removed, copy-pasta

File: coderd/database/querier_test.go (hunk 3 above)
Comment (Emyrk) [reply coderd/database/querier_test.go:795]: I will comment, but yes :+1:

Comment (Emyrk) [reply coderd/database/querier_test.go:795]: I do above in [`siteWideIDs`](https://github.com/coder/coder/blob/f45156c062bfb16a7371407c3fedc4515e87957c/coderd/database/querier_test.go#L784-L784)

**Review by johnstcn — APPROVED** [review]

**Pull request merged**